lib/gc/Transforms/Pipeline.cpp

```diff
@@ -0,0 +1,166 @@
+  bufferization::BufferResultsToOutParamsOpts opt{};
+  opt.hoistStaticAllocs = true;
+  pm.addPass(bufferization::createBufferResultsToOutParamsPass(opt));
+  // + buffer schedule pass, down-stream pass, to migrate buffer reschedule pass
+  // from GC V1.
+  pm.addNestedPass<func::FuncOp>(
+      bufferization::createBufferHoistingPass()); // Need to improve this pass
+                                                  // to support thread-local
+                                                  // allocator.
+  pm.addNestedPass<func::FuncOp>(bufferization::createBufferLoopHoistingPass());
+  pm.addNestedPass<func::FuncOp>(bufferization::createBufferDeallocationPass());
+  pm.addPass(createBufferizationToMemRefPass());
+}
+// scf + arith + math + vector + memref + linalg.brgemm
+
+void populateMicroKernelPasses(mlir::PassManager &pm) {
+  // + ConvertLinalgToMicrokernel pass, upstream-able,
+  // + CleanupInvalidMicrokernel pass, upstream-able
+  // + InvariantMicrokernelMotion pass, upstream-able
+  // + ConvertMicrokernelToDnnlFunc, down-stream pass, to lower brgemm to dnnl
+  // call
+  // + ConvertMicrokernelToXsmm, down-stream pass, to lower brgemm to libxsmm
+  // call
+  // + LowerMicrokernel pass, upstream-able
+  // + DispatchMicrokernel, down-stream pass
+}
+// scf + arith + math + vector + memref + func/microkernel
+
+void populateCPURuntimePasses(mlir::PassManager &pm) {
+  // + flatten nested parallel pass, down-stream pass, to support coarse-grain
+  // fusion
+  // remove this pass after we add FlattenNestedParallel
+  pm.addPass(createConvertSCFToOpenMPPass());
+}
+
+void populateLoweringToLLVMPasses(mlir::PassManager &pm) {
+  pm.addPass(createConvertSCFToCFPass());
+  pm.addPass(cpuruntime::createCPURuntimeToLLVM());
+  pm.addPass(createConvertOpenMPToLLVMPass());
+  pm.addNestedPass<func::FuncOp>(createConvertMathToLLVMPass());
+  pm.addPass(createConvertMathToLibmPass());
+  pm.addPass(createFinalizeMemRefToLLVMConversionPass());
+  pm.addNestedPass<func::FuncOp>(createArithToLLVMConversionPass());
+  pm.addPass(createConvertFuncToLLVMPass());
+  pm.addPass(createConvertControlFlowToLLVMPass());
+  pm.addPass(createCSEPass());
+  pm.addPass(createCanonicalizerPass());
+  pm.addPass(createReconcileUnrealizedCastsPass());
+  pm.addPass(createSymbolDCEPass());
+}
+
+void populateLLVMPasses(mlir::PassManager &pm) {
+  pm.addPass(memref::createExpandOpsPass());
+  pm.addPass(memref::createExpandStridedMetadataPass());
+  populateLoweringToLLVMPasses(pm);
+}
+
+void populateCPUPipeline(mlir::PassManager &pm) {
+  // front-end, oneDNN graph dialect
+  populateFrontendPasses(pm);
+  // middle-end, LinalgX/Linalg/tensor dialects
+  populateTensorPasses(pm);
+  // middle-end, arith/math/vector dialects
+  populateVectorPasses(pm);
+  // back-end, arith/math/vector/memref dialects
+  populateBufferizationPasses(pm);
+  // REMOVE this pass after the TensorPasses are added. Currently we add this
```

Comment:
Can we add "TODO" to highlight the remaining items to better distinguish with other common comments?